### DISCUSSION TITLE
ODBC connect with a MariaDB server

### DISCUSSION THREAD

**Discussion by Brandon-sistein** [frangoteam/FUXA#1446] [Q&A] [answered]
Hello, I'm trying to conect fuxa with a MariaDB server and show me this error:

"getendpoints-error: Error: [odbc] Error getting table information"

I tried with this configuration:

`DRIVER=MySQL;SERVER=192.168.0.125;PORT=3307;DATABASE=Brandolinera`

When I watch the device show me "online":

![image](https://github.com/user-attachments/assets/49b2f2cf-ed3d-46cd-8219-ee317930de11)

**Comment by MatthewReed303** [comment]
Hi @Brandon-sistein can you try with `SSLMODE=DISABLED` as I have to use this to get mySQL to connect

`DRIVER=MySQL;SERVER=Your_DB_IP;PORT=3306;DATABASE=testDB;SSLMODE=DISABLED`

Are you using the Docker version? 

Sometimes I get the same error but the connection does work, give it a go from the script 

https://github.com/frangoteam/FUXA/wiki/HowTo-ODBC

**Reply by Brandon-sistein** [reply]
Still showing the error, I will try to use the script

I have docker version

**Reply by Brandon-sistein** [reply]
Hi @MatthewReed303 first of all thanks for helping me.
The problem still here. I made this script:

```
// Initialize the device, sane name as connection
let myDevice = await $getDevice('MariaDB', true);

let result = await myDevice.pool.query(`SELECT * FROM Brandolinera.cars;`);
console.log(JSON.stringify(result));
```

And show me this error:

>[object Object]
>"{\"odbcErrors\":[{\"state\":\"08S01\",\"code\":2013,\"message\":\"[MySQL][ODBC 8.4(w) Driver][mysqld-5.5.5-10.5.8-MariaDB-log]Lost connection to MySQL server during query\"}]}


The query works good in node red:

![image](https://github.com/user-attachments/assets/59b42528-a215-4c31-9c5d-a36437f9b48b)

Any idea to solve it?

I was thinking to solve it taking the data using node red and mqtt but it is a bungle in my opinion

**Reply by Brandon-sistein** [reply]
@MatthewReed303 it is solved the problem is that I have to active the SSLMODE I will put the correct way to configure in this post

**Comment by Brandon-sistein** [comment, marked as answer]
I solved with this configuration:

`DRIVER=MySQL;SERVER=192.168.0.125;PORT=3307;DATABASE=Brandolinera;SSLMODE=ENABLE`
